### PR TITLE
Gdr 1955

### PR DESCRIPTION
**Pull request by bczech**
# Description
## What changed?
Refactored normalization types
Related JIRA issue: GDR-1955

## Why was it changed?
To unify normalization_types in all assays

# Checklist for sustainable code base
- [ ] I added tests for any code changed/added
- [ ] I added documentation for any code changed/added
- [ ] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [X] Package version bumped
- [X] Changelog updated

# Screenshots (optional)

### CODE REVIEW

File: R/combinations-calculate_matrix_metric.R [hunk 1]
```diff
@@ -92,18 +92,15 @@ calculate_Bliss <- function(sa1, series_id1, sa2, series_id2, metric) {
   checkmate::assert_true(all(sa1[[series_id2]] == 0L))
```
Comment (j-smola) [inline R/combinations-calculate_matrix_metric.R:92]: Param `metric` seems to be not needed.

OR You change `metric = "average"` (and adapt the rest of the code accordingly)

**Review by j-smola — APPROVED** [review]

File: R/combinations-isobolograms.R [hunk 2]
```diff
@@ -338,11 +338,11 @@ calculate_Loewe <- function(
 }
 
 
-get_isocutoffs <- function(df_mean, normalization_type) {
-  if (min(df_mean[[normalization_type]], na.rm = TRUE) > 0.7) {
+get_isocutoffs <- function(df_mean, normalization_type, data_col = "average") {
```
Comment (j-smola) [inline R/combinations-isobolograms.R:341]: Is there any reason why all the functions in `combinations-isobologram.R` have no tests?

File: R/combinations-calculate_matrix_metric.R [hunk 3]
```diff
@@ -92,18 +92,15 @@ calculate_Bliss <- function(sa1, series_id1, sa2, series_id2, metric) {
...
   checkmate::assert_true(all(sa2[[series_id1]] == 0L))
```
[the remark this answers is not in the document]
Comment (bczech) [reply R/combinations-calculate_matrix_metric.R:94]: Ok, corrected.

File: R/combinations-calculate_matrix_metric.R (hunk 1 above)
Comment (bczech) [reply R/combinations-calculate_matrix_metric.R:92]: Removed

File: R/combinations-calculate_matrix_metric.R [hunk 4]
```diff
@@ -92,18 +92,15 @@ calculate_Bliss <- function(sa1, series_id1, sa2, series_id2, metric) {
...
 
   idx <- match(u[[series_id1]], sa1[[series_id1]])
   
-  u <- base::merge(u, sa1[, c(..series_id1, "metric1")], by = series_id1)
-  u <- base::merge(u, sa2[, c(..series_id2, "metric2")], by = series_id2)
+  u <- base::merge(u, sa1[, c(..series_id1, "average")], by = series_id1)
+  u <- base::merge(u, sa2[, c(..series_id2, "average")], by = series_id2)
 
-  metric <- do.call(FXN, list(u$metric1, u$metric2))
+  metric <- do.call(FXN, list(u$average.x, u$average.y))
```
[the remark this answers is not in the document]
Comment (bczech) [reply R/combinations-calculate_matrix_metric.R:104]: Metric param was removed

File: R/fit_SE.combinations.R [hunk 5]
```diff
@@ -237,30 +233,30 @@ fit_SE.combinations <- function(se,
         colnames(complete), 
         c(metric, "row_values", "col_values", "codil_values")
       )
-      mat <- as.matrix(complete[, ..keep])
-      complete$average <- rowMeans(mat, na.rm = TRUE)
+      complete$average <- rowMeans(complete[, ..keep], na.rm = TRUE)
```
[the remark this answers is not in the document]
Comment (bczech) [reply R/fit_SE.combinations.R:237]: Done

File: R/fit_SE.combinations.R [hunk 6]
```diff
@@ -237,30 +233,30 @@ fit_SE.combinations <- function(se,
...
       if (NROW(av_matrix) == 0) {
         av_matrix <- h_excess <- b_excess <- NULL
       } else {
         
-        cols <- c(id, id2, metric)
-        sa1 <- av_matrix[av_matrix[[id2]] == 0, ..cols]
-        sa2 <- av_matrix[av_matrix[[id]] == 0, ..cols]
+        cols <- c(id, id2, "average")
+        sa1 <- av_matrix[get(id2) == 0, ..cols]
+        sa2 <- av_matrix[get(id) == 0, ..cols]
```
[the remark this answers is not in the document]
Comment (bczech) [reply R/fit_SE.combinations.R:252]: Done

File: R/combinations-isobolograms.R (hunk 2 above)
Comment (bczech) [reply R/combinations-isobolograms.R:341]: Indeed, there are no tests. I will try to create some even basic tests for this function

**Pull request merged**